dbt-tests-adapter/dbt/tests/adapter/constraints/fixtures.py

```diff
@@ -363,7 +363,8 @@
           - type: check
             expression: id >= 1
           - type: foreign_key
-            expression: {schema}.foreign_key_model (id)
+            to: ref('foreign_key_model')
```

Comment:
Are we removing `expression` then? If so, is this potentially breaking? If not, do we need a test that still captures scenarios where we use `expression`?

Reply:
"expression" isn't being removed, the test was just changed to use "to:" instead.

Reply:
"expression" backward compatibility is still preserved, and there is unit test coverage for that. Personally I don't think it's worth the lift to introduce an integration test across all adapters given that `to` and `to_columns` will be the recommended way to use foreign key constraints going forward, and that we already have more thorough edge case handling for `expression` in unit testing.

Reply:
here's where the back-compat is implemented: 
* model-level: https://github.com/dbt-labs/dbt-adapters/pull/260/files#diff-e2306b841793e5b07177010e8920c59163df1536306fc76e7f6a09d36dcae4d3R1705-R1706
* column-level: https://github.com/dbt-labs/dbt-adapters/pull/260/files#diff-e2306b841793e5b07177010e8920c59163df1536306fc76e7f6a09d36dcae4d3R1606-R1607

Reply:
I found the unit tests. I think that's fine. I get hesitant when we remove tests for supported functionality, even for deprecated functionality, because that's where we wind up with regressions. But the rationale makes sense. Should we have a deprecation warning on the use of `expression` then?

Reply:
We're not deprecating or planning to deprecate `expression` in the future, so I don't think a deprecation warning makes sense to introduce. `expression` will still be useful for other constraint types, and can serve as a more fleible fallback for more complex rendering use cases, or perhaps for adapters where the default 'to' and 'to_columns' implementation is not compatible + has not been overwritten by the adapter maintainer yet. Likely the way we'll document this is that `to` and `to_columns` are the recommended way to define foreign key constraints, and also describe the expression option.

Reply:
Ok, I think I'm convinced of this then. `expression` is still a plausible use case because `to` / `to_columns` only covers 90% of the use cases with `expression` as a backup. We also need to account for `expression` because we don't want to break third party adapters.